### ISSUE TITLE
`vue/define-props-destructuring` erroring when no assignment occurs

### ISSUE THREAD

**Issue by simon-tma** [vuejs/eslint-plugin-vue#3006]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.39.2
- **eslint-plugin-vue version:** 10.7.0
- **Vue version:** 3.5.26
- **Node version:** 24
- **Operating System:** macOS 

**Please show your full configuration:**

```json5
rules: {
    'vue/define-props-destructuring': 'error'
}
```

**What did you do?**	

```vue
<script setup lang="ts">
defineProps<{ message: string }>();
</script>
```

**What did you expect to happen?**

No setting for `vue/define-props-destructuring` should generate an error when props aren't accessed within `<script setup>`

**What actually happened?**
```
error  Prefer destructuring from `defineProps` directly  vue/define-props-destructuring
```

**Comment by FloEdelmann** [comment]
This was already mentioned in https://github.com/vuejs/eslint-plugin-vue/issues/2688#issuecomment-2841926404 and https://github.com/vuejs/eslint-plugin-vue/issues/2688#issuecomment-2850439183 (but neither of those two commenters bothered enough to open an issue).

I like @esfahanweb's suggestion there: Keep the current behavior as `destructure: "always"`, introduce a new option `destructure: "only-when-assigned"` and maybe make that the default.
